tests/cases/compiler/scopeCheckStaticInitializer.ts

```diff
@@ -0,0 +1,14 @@
+class X {
```

Comment:
The bug can also be observed on non-statics:

```
class Y {
  a = "" + this.b;
  b = "";
}
```

I'm not sure if your fix handles this case as well.